tests/gnrc_tcp/Makefile

```
@@ -0,0 +1,66 @@
+include ../Makefile.tests_common
+
+# Basic Configuration
+BOARD ?= native
+TAP ?= tap0
+
+# Shorten default TCP timeouts to speedup testing
+MSL_US ?= 1000000
+TIMEOUT_US ?= 3000000
+
+# Mark Boards with insufficient memory
+BOARD_INSUFFICIENT_MEMORY := airfy-beacon arduino-duemilanove arduino-leonardo arduino-mega2560 \
+                             arduino-nano arduino-uno arduino-duemilanove b-l072z-lrwan1 hifive1 \
+                             hifive1b i-nucleo-lrwan1 lsn50 microbit msb-430 msb-430h nrf51dongle \
+                             nrf6310 nucleo-f030r8 nucleo-f031k6 nucleo-f042k6 nucleo-f070rb \
+                             saml10-xpro saml11-xpro stm32f0discovery stm32l0538-disco telosb \
+                             nucleo-f072rb nucleo-f303k8 nucleo-f334r8 nucleo-l031k6 \
+                             nucleo-l053r8 mega-xplained waspmote-pro wsn430-v1_3b wsn430-v1_4 \
+                             yunjia-nrf51822 z1 thingy52
+
+# unsupported by ethos: chronos, hamilton, ruuvitag
+BOARD_BLACKLIST := chronos hamilton ruuvitag
+
+# This test depends on tap device setup (only allowed by root)
+# Suppress test execution to avoid CI errors
+TEST_ON_CI_BLACKLIST += all
+
+CFLAGS += -DSHELL_NO_ECHO
+CFLAGS += -DGNRC_TCP_MSL=$(MSL_US)
+CFLAGS += -DGNRC_TCP_CONNECTION_TIMEOUT_DURATION=$(TIMEOUT_US)
+
+USEMODULE += gnrc_netdev_default
```

```suggestion
ifeq (native,$(BOARD))
  USEMODULE += netdev_tap
  TERMFLAGS ?= $(TAP)
else
  USEMODULE += stdio_ethos

  ETHOS_BAUDRATE ?= 115200
  CFLAGS += -DETHOS_BAUDRATE=$(ETHOS_BAUDRATE)
  TERMDEPS += ethos
  TERMPROG ?= sudo $(RIOTTOOLS)/ethos/ethos
  TERMFLAGS ?= $(TAP) $(PORT) $(ETHOS_BAUDRATE)
endif
```

(And of course remove the `ifeq (,$(filter native,$(BOARD)))` block below). Otherwise you pull in the default network device for the device, which

a) wastes memory since it is not used
b) might block `ethos` from initialization (and having it run into [an assertion](https://github.com/RIOT-OS/RIOT/blob/abdebaa86f375a567a4735acd001e758f131140d/sys/net/gnrc/netif/gnrc_netif.c#L64), since `GNRC_NETIF_NUMOF == 1`